docs/workflows/kserve-predictor.md

```diff
@@ -13,12 +13,125 @@ The KServe predictor step expects a model URL to be supplied as input, pointing
+    outputs:
+      - name: prediction-url
+    extensions:
+      - name: mlflow-s3-store
+        product: mlflow
+        service_resource: s3
+      - name: kserve
+        service_resource: kserve-api
+```
+
+Aside from the mandatory input `model` parameter that needs to be a URL pointing to the location of a trained ML model saved in an MLflow artifact store, the KServe predictor workflow step also accepts the following optional input parameters that can be used to customize how the prediction service is created and updated:
+
+- `predictor`: this can be used to configure the type of KServe predictor engine used for the prediction service. This can take the following values:
+
+    - if set to `auto`, the KServe predictor will automatically detect the type of model from the MLflow metadata present in the artifact store and use the appropriate predictor engine.
+    - `tensorflow`: use to serve models with the TensorFlow Serving engine. Only works with models trained with TensorFlow or Keras and saved using the TensorFlow saved_model model format.
+    - `sklearn`: use to serve models trained with scikit-learn and saved in the sklearn pickled model format
+    - `triton`: use the NVidia Triton prediction back-end. Works with models trained with TensorFlow or Keras and saved using the TensorFlow saved_model format and with models in ONNX format.
+
+- `app_name`: use this to explicitly set the name of the FuseML application used to represent the KServe prediction service. Its value also determines the prediction URL as well as the names of the Kubernetes resources created by the KServe predictor. Our example uses an expression to dynamically set the `app_name` parameter to the name and project of the MLflow codeset used as workflow input. If not set, the application name is constructed by combining the workflow name with the name of an input codeset name and project, if one is provided as input. In the absence of an input codeset, the application name is generated by concatenating the workflow name with a randomly generated string.
+
+    !!! note
+
+        Choosing a value for the `app_name` parameter should be done with care, as it is used to uniquely identify a FuseML application and its associated Kubernetes resources (i.e. the name of the KServe prediction service, prediction URL etc.). It can lead to a situation where the same KServe prediction service is managed by more than one FuseML workflows. In this case, the results can be unpredictable, because multiple workflows will compete over managing the same application.
+        
+    !!! warning
+    
+        If an `app_name` value is not provided and the predictor step doesn't receive an input codeset, the generated application name will be random, which means that every workflow run will create a new application and prediction service. This should be avoided, as it easily lead to resource exhaustion.
```

Comment:
Are we actually able to run a workflow without a codeset?

Reply:
yes, a workflow doesn't have to have a codeset specified as input and used in any of its steps. to run it, you still need to assign it a codeset, but otherwise everything works just the same